libs/logging/src/test_utils.ts

```diff
@@ -3,6 +3,6 @@ import { LogSource } from './log_source';
 
 export function fakeLogger(logSource: LogSource = LogSource.System): Logger {
   const logger = new Logger(logSource);
-  jest.spyOn(logger, 'log').mockResolvedValue();
+  logger.log = jest.fn().mockResolvedValue(undefined);
```

Comment:
With the spy approach, I was running into errors from `console.log` being called after tests completed (I think because the state machine kept running for some reason). It seemed like the spying wasn't actually replacing the real implementation of `logger.log`, so I decided to just change to a straight up mock.